commit 2fc47feff98ca5a0c1dcc192df840efd0b28027e (refs/changes/17/2517/3)
Author: Andrey Yatskov <andrey.yatskov@silvertreesystems.com>
Date:   2017-09-22T13:51:52-07:00 (2 years, 1 month ago)
    
    ADMINUI-2378 Adminui should trim leading and trailing whitespace from search input fields
    Reviewed by: Angela Fong <angela.fong@joyent.com>
    Approved by: Angela Fong <angela.fong@joyent.com>

diff --git a/lib/adminui.js b/lib/adminui.js
index 768ab268..d0c47d6e 100644
--- a/lib/adminui.js
+++ b/lib/adminui.js
@@ -32,12 +32,20 @@ var notes = require('./notes');
 var settings = require('./settings');
 
 function resume(req, res, next) {
+
     process.nextTick(function () {
         req.resume();
     });
     next();
 }
 
+function paramsTrimmer(req, res, next) {
+    Object.keys(req.params).forEach(function (paramName) {
+        req.params[paramName] = req.params[paramName].trim();
+    });
+    next();
+}
+
 function setUfds(req, res, next) {
     var trySetUfds = function () {
         if (req.ufdsMaster.connected) {
@@ -147,6 +155,7 @@ ADMINUI.prototype.createServer = function () {
     server.use(restify.queryParser());
     server.use(restify.gzipResponse());
     server.use(restify.acceptParser(server.acceptable));
+    server.use(paramsTrimmer);
 
     TraceEvent.mount({
         skipRoutes: ['ping', 'getca'],
